From eb3c0ac68b3a47bdb16b5e44e04bed19d8d3aa17 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
Date: Thu, 6 Nov 2014 09:42:08 +0100
Subject: [PATCH] usb: android: provide kobj reference for f_acm

So that function driver can provide sysfs nodes for instance
specific customizations.

Change-Id: I80ff7766a28eae8437b2bf65189bfc450520bbb2
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
---
 drivers/usb/gadget/android.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index bf66f9c..86aec5b 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -479,6 +479,11 @@ acm_function_init(struct android_usb_function *f,
 			ret = PTR_ERR(config->f_acm_inst[i]);
 			goto err_usb_get_function_instance;
 		}
+
+		/* Provide kobject reference */
+		if (!config->f_acm_inst[i]->fd->parent)
+			config->f_acm_inst[i]->fd->parent = &f->dev->kobj;
+
 		config->f_acm[i] = usb_get_function(config->f_acm_inst[i]);
 		if (IS_ERR(config->f_acm[i])) {
 			ret = PTR_ERR(config->f_acm[i]);
-- 
1.9.1

